### ISSUE TITLE
Redirect oauth user back to original page after email confirmation

### ISSUE THREAD

**Issue by christoph-buente** [openstreetmap/openstreetmap-website#895]
For wheelmap.org we allow users to login via oauth with a valid openstreetmap.org account. But new users, who sign up with wheelmap (and consequently with openstreetmap) end on a welcome page after they click the confirmation page in their email.

http://www.openstreetmap.org/welcome

This behavior might be intended for users who signed for openstreetmap directly, but wheelmap users (and possibly other users of third party services using oauth) are pretty much lost.

Is there a way to redirect these users to the original website after they clicked the confirmation link? Maybe by passing an additional parameter when sending them to the osm login page?

That would be very helpful. I could possibly provide a pull request after a discussion.

**Comment by tomhughes** [comment]
They should get sent to whatever callback URL you specified in your oauth requests - there are tests for that I believe.

**Comment by tomhughes** [comment]
https://github.com/openstreetmap/openstreetmap-website/blob/master/test/integration/oauth_test.rb#L225 is the test in question.

**Comment by HolgerJeromin** [comment]
i think the point is about new users, which has to create a new OSM account first. Not only authorise them.

**Comment by tomhughes** [comment]
Right, but they should still end up going through oauthorize at the end right, to authorize the new account?

**Comment by christoph-buente** [comment]
Yes, they need to finish the authorize process and end up on the page they came from.

**Comment by tomhughes** [comment]
My point was that when the get to `oauthorize` the callback URL should still work and send them back to your site.

**Comment by pietervdvn** [comment]
Hey, this is still an issue.

The flow is:

1. A user (without OSM-account) browses an app (e.g. wheelmap.org or mapcomplete.org)
2. They want to edit a feature, so they click 'log in'
3. They create a new account and land on the 'verify your email'-step
4. They open their email client, find the email and click the 'verify my email address link'
5. The 'verify my email address' redirects to the OSM-homepage, which they have never seen before... This should instead take them back to the original webpage or application.

(I made a quick screencap here: https://imgur.com/a/sCZJaZA):

Part two is especially interesting: https://i.imgur.com/RUFLQof.mp4

**Comment by tomhughes** [comment]
What we really need is the full list of URLs that are visited so that we can tell where the callback is getting list - in theory it should be preserved through the signup process but that is notoriously complex.
